### PR TITLE
Add linkinator link checking

### PR DESCRIPTION
**Pull request by Noah-Silvera**
- Add the linkinator package to check all links in the generated build content
- Move the spell check before the external link checking
- Fix existing broken links

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -5,11 +5,11 @@
   "requires": true,
   "dependencies": {
     "@babel/code-frame": {
-      "version": "7.10.1",
-      "resolved": "https://registry.npmjs.org/@babel/code-frame/-/code-frame-7.10.1.tgz",
-      "integrity": "sha512-IGhtTmpjGbYzcEDOw7DcQtbQSXcG9ftmAXtWTu9V936vDye4xjjekktFAtgZsWpzTj/X01jocB46mTywm/4SZw==",
+      "version": "7.8.3",
```
Comment (ianschmitz) [inline package-lock.json:8]: Looks like a lot of packages that were updated in my last PR were downgraded here?

Maybe try reverting this file and adding your package again after deleting `node_modules`.

File: package.json [hunk 2]
```diff
@@ -5,8 +5,10 @@
   "scripts": {
     "start": "docusaurus start",
     "build": "docusaurus build",
-    "lint": "npm run lint:links && npm run lint:spelling && npm run lint:style",
-    "lint:links": "remark -f -q -u validate-links \"docs/**/*.mdx\"",
+    "lint": "npm run lint:spelling && npm run lint:links && npm run lint:style",
```
Comment (ianschmitz) [inline package.json:8]: Good call on changing the order. Maybe put links at the end so prettier will fail before it does the external requests as part of `linkinator`?

File: package.json [hunk 3]
```diff
@@ -5,8 +5,10 @@
...
+    "lint:links": "npm run lint:links:md && npm run lint:links:ext",
+    "lint:links:md": "remark -f -q -u validate-links \"docs/**/*.mdx\"",
+    "lint:links:ext": "npm run build && linkinator ./build -r --silent --concurrency 10",
```
Comment (ianschmitz) [inline package.json:11]: I don't think we should require `build` to have been run here. It doubles our build time which is around ~50s in CI.

File: package.json (hunk 2 above)
Comment (Noah-Silvera) [reply package.json]: Moved prettier to before links!

File: package.json (hunk 3 above)
Comment (Noah-Silvera) [reply package.json:11]: If we don't require build to be run, then how can we be sure we are checking the latest up to date website's external links?

Comment (ianschmitz) [reply package.json:11]: `build` is run before `lint` in CI. Is that what you mean?

Comment (Noah-Silvera) [reply package.json]: doh, my bad. I was thinking local. Of course build runs before lint in CI.

Removed the `npm run build ` call from the lint step.

**Pull request merged**